x/staging/posts/keeper/msgs_server.go

```diff
@@ -131,7 +130,7 @@ func (k msgServer) EditPost(goCtx context.Context, msg *types.MsgEditPost) (*typ
 
 	// Check the validity of the current block height respect to the creation date of the post
 	if existing.Created.After(ctx.BlockTime()) {
-		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "edit date cannot be before creation date")
+		return nil, types.ErrInvalidEditDate
```

Comment:
This is related to the request so we should use a request-related error
```suggestion
		return nil, sdkerrors.Wrap(sdkerrors.ErrInvalidRequest, "edit date cannot be before creation date")
```